### PR TITLE
BIGTOP-4148. Replace adding repo in puppetize.sh with puppet manifest

### PR DESCRIPTION
**Pull request by masatana**
### Description of PR
https://issues.apache.org/jira/browse/BIGTOP-4148

### How was this patch tested?

Run the commands below on both Rocky 8/Rocky 9:

```
$ sudo dnf install git -y
$ sudo yum install java-1.8.0-openjdk-devel -y
$ git clone -b BIGTOP-4148 https://github.com/masatana/bigtop.git
$ cd bigtop
$ sudo bigtop_toolchain/bin/puppetize.sh
$ ./gradlew toolchain
```

And confirmed the Debug log like below:

Rocky8

```
(...snip...)
Debug: /Stage[main]/Bigtop_toolchain::Packages/Yumrepo[powertools]/before: before to Package[pandoc]
Debug: /Stage[main]/Bigtop_toolchain::Packages/Yumrepo[devel]/before: before to Package[java-11-openjdk-devel]
(...snip...)
Notice: /Stage[main]/Bigtop_toolchain::Packages/Yumrepo[powertools]/enabled: enabled changed '0' to '1'                     Debug: /Stage[main]/Bigtop_toolchain::Packages/Yumrepo[powertools]: The container Class[Bigtop_toolchain::Packages] will propagate my refresh event                                                                                                     Notice: /Stage[main]/Bigtop_toolchain::Packages/Yumrepo[devel]/enabled: enabled changed '0' to '1'  
(...snip...)
```

Rocky9
```
(...snip...)
Debug: /Stage[main]/Bigtop_toolchain::Packages/Yumrepo[crb]/before: before to Package[pandoc]
Debug: /Stage[main]/Bigtop_toolchain::Packages/Yumrepo[devel]/before: before to Package[java-11-openjdk-devel]
(...snip...)
Debug: Prefetching inifile resources for yumrepo                                                                            Notice: /Stage[main]/Bigtop_toolchain::Packages/Yumrepo[crb]/enabled: enabled changed '0' to '1'
Debug: /Stage[main]/Bigtop_toolchain::Packages/Yumrepo[crb]: The container Class[Bigtop_toolchain::Packages] will propagate my refresh event                                                                                                            Notice: /Stage[main]/Bigtop_toolchain::Packages/Yumrepo[devel]/enabled: enabled changed '0' to '1'
(...snip...)
```

### For code changes:

- [x] Does the title or this PR starts with the corresponding JIRA issue id (e.g. 'BIGTOP-3638. Your PR title ...')?
- [x] Make sure that newly added files do not have any licensing issues. When in doubt refer to https://www.apache.org/licenses/

### CODE REVIEW

File: bigtop_toolchain/manifests/packages.pp [hunk 1]
```diff
@@ -348,6 +348,37 @@
...
+        ensure  => 'present',
+        enabled => '1'
+      }
+      yumrepo { 'devel':
+        ensure  => 'present',
+        enabled => '1'
+      }
+      Yumrepo<||> -> Package<||>
+    }
+  }
```
Comment (iwasakims) [inline bigtop_toolchain/manifests/packages.pp:381]: @masatana Could you add equivalent of this to the manifest of bigtop-deploy too? Some packages (e.g. python3-devel on rockylinux-8) seem to need this. We are using bigtop-deploy over bigtop/puppet images on CI.

**Comment by masatana** [comment]
Updated the patch, and did some addendum tests.

## Rocky 8
```
$ ./gradlew -POS=rockylinux-8 bigtop-puppet
$ cd provisioner/docker
$ ./docker-hadoop.sh -dcp -F docker-compose-cgroupv2.yml -C config_rockylinux-8.yaml -m 4g -c 3 -k "hdfs"
(snip)
Notice: /Stage[main]/Bigtop_repo/Yumrepo[Bigtop_0]/ensure: created
Notice: /Stage[main]/Yumrepo/Yumrepo[powertools]/enabled: enabled changed '0' to '1'
Notice: /Stage[main]/Yumrepo/Yumrepo[devel]/enabled: enabled changed '0' to '1'
Notice: Baseurl: http://repos.bigtop.apache.org/releases/3.2.1/rockylinux/8/$basearch 
(snip)
Notice: /Stage[main]/Hadoop::Init_hdfs/Exec[init hdfs]/returns: executed successfully                                       Notice: Applied catalog in 114.29 seconds     
```

## Rocky 9
```
$ ./gradlew -POS=rockylinux-9 bigtop-puppet
$ cd provisioner/docker
$ ./docker-hadoop.sh -dcp -F docker-compose-cgroupv2.yml -C config_rockylinux-9.yaml -m 4g -c 3 -k "hdfs"
(snip)
Notice: Compiled catalog for 2d52d108752b.bigtop.apache.org in environment production in 0.45 seconds
Notice: /Stage[main]/Yumrepo/Yumrepo[crb]/enabled: enabled changed '0' to '1'
Notice: /Stage[main]/Yumrepo/Yumrepo[devel]/enabled: enabled changed '0' to '1'
Notice: Baseurl: http://repos.bigtop.apache.org/releases/3.2.1/rockylinux/9/$basearch 
(snip)
(As of right now, this command fails as there is no rpm for rocky linux 9 in bigtop-3.2.1)
```

File: bigtop_toolchain/manifests/packages.pp (hunk 1 above)
Comment (masatana) [reply bigtop_toolchain/manifests/packages.pp:381]: @iwasakims Thanks for the comment. Could you check the updated patch?

**Review by iwasakims — APPROVED** [review]
LGTM. Thanks, @masatana.

**Pull request merged**